### PR TITLE
chore: 🤖 Migrate `form.inpu` to use Hds::TextInput::Field`

### PR DESCRIPTION
**Pull request by dvonanderson**
✅ Closes: https://hashicorp.atlassian.net/browse/ICU-12466
✅ Closes: https://hashicorp.atlassian.net/browse/ICU-12467
✅ Closes: https://hashicorp.atlassian.net/browse/ICU-12468
✅ Closes: https://hashicorp.atlassian.net/browse/ICU-12470

:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-12466)
:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-12467)
:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-12468)
:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-12470)

## Description
- Migrate `form/group`, `form/host-static`, `form/host-catalog/aws` and `form/host-catalog/azure` forms to use `Hds::TextInputField instead of `form.input`
- Fix corresponding test

## Screenshots (if appropriate):
- NOTE: To many screenshots.

## How to Test
- Click on Vercel link for Admin UI or pull down PR and run it locally using yarn start
- Click on the "Orgs" in sidebar navigation and choose any org
- Click on any project card
    - FOR TESTING GROUPS
        - Click on "New Group" and ensure that when you click on a type card that all input fields are represented and inputed data is able to be saved.
    - FOR STATIC HOST
        - Click on "New Host Catalog" and click the "Static" card and ensure that all input fields are represented and inputed data is able to be saved.
    - FOR DYNAMIC HOST
        - Click on "New Host Catalog" and click the "Dynamic" card and choose either "aws" or "azure" card and ensure that all input fields are represented and inputed data is able to be saved.

:technologist: [Admin preview](PATH_TO_FEATURE)

## Checklist:

~~- [ ] I have added before and after screenshots for UI changes~
~~- [ ] I have added JSON response output for API changes~~
~~- [ ] I have added steps to reproduce and test for bug fixes in the description~~
~~- [ ] I have commented on my code, particularly in hard-to-understand areas~~
- [x] My changes generate no new warnings
~~- [ ] I have added tests that prove my fix is effective or that my feature works~~

### CODE REVIEW

File: ui/admin/tests/acceptance/host-catalogs/hosts/update-test.js [hunk 1]
```diff
@@ -130,7 +130,7 @@ module('Acceptance | host-catalogs | hosts | update', function (hooks) {
       'The request was invalid.',
     );
     assert.ok(
-      find('.rose-form-error-message').textContent.trim(),
+      find('.hds-form-error__message').textContent.trim(),
```
Comment (DhariniJeeva) [inline ui/admin/tests/acceptance/host-catalogs/hosts/update-test.js:133]: Can we start using data-attrs here like @cameronperera mentioned in tech time?

Comment (dvonanderson) [reply ui/admin/tests/acceptance/host-catalogs/hosts/update-test.js:133]: @DhariniJeeva Absolutely!

**Review by lisbet-alvarez — APPROVED** [review]
🙌

**Review by calcaide — APPROVED** [review]
Nice migration! :raised_hands:

**Pull request merged**